packages/flutter/lib/src/painting/text_editing.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Whether a [TextPosition] is visually upstream or downstream of its offset.
+///
+/// When a text position exists at a line break, a single offset has two visual
+/// positions, one prior to the line break (at the end of the first line) and
+/// one after the line break (at the start of the second line). A text affinity
+/// disambiguates between those cases.
+enum TextAffinity {
+  /// The position has affinity for the upstream side of the text position.
+  ///
+  /// For example, if the offset of the text position is a line break, the
```

Comment:
"a line break" is ambiguous here. Do you mean before or after the U+000A character, for instance?